src/analysis/retail/evoker/augmentation/modules/features/BuffTargetHelper/BuffTargetHelper.tsx

```diff
@@ -0,0 +1,339 @@
+ * But knowing who to buff and when can be tricky to figure out at a glance.
+ * Given multiple factors such as time intervals (Ebon Might timings), filtering
+ * off damage that isn't amplified by buffs, intimate knowledge about different specs
+ * where they perform, where they don't, player skill, etc.
+ *
+ * The aim for this module is to streamline the process, by providing a list of the targets
+ * that would have been best to buff in a given time interval.
+ *
+ * Along with this we will provide the user with a MRT note export that can be used with
+ * various WA's to help keep track of who to Prescience.
+ *
+ * We don't provide analysis on actual gameplay in this module, this is strictly meant
+ * to assist people in making more informed decisions about who to buff at what times in
+ * a given boss fight.
+ *
+ * Hopefully with this information being easily available, players will be able to use this
+ * data and predict who will be high performing target on other fights as well.
+ * (Assuming you are playing with the same people again eg. reclears/prog).
+ */
+class BuffTargetHelper extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+  protected combatants!: Combatants;
+
+  // This could be set to fixed 30s intervals, but might as well give some attention to the Interwoven gamers!
+  interval: number =
+    30 * 1000 * (this.selectedCombatant.hasTalent(TALENTS.INTERWOVEN_THREADS_TALENT) ? 0.9 : 1);
+
+  fightStart: number = this.owner.fight.start_time;
+  fightEnd: number = this.owner.fight.end_time;
+  mrtPrescienceHelperNote: string = '';
+
+  constructor(options: Options) {
+    super(options);
+    /** No need to show this in dungeon runs, for obvious reasons */
+    this.active = !isFightDungeon(this.owner.fight);
+
+    /** Populate our whitelist */
+    this.addEventListener(Events.fightend, () => {
+      /** Clear the maps, we do this since otherwise it has a tendency
+       * to keep the values from a previous fight if you swap around without
+       * refreshing the browser */
+      playerDamageMap.clear();
+      playerWhitelist.clear();
+
+      const players = Object.values(this.combatants.players);
+      players.forEach((player) => {
+        if (!player.spec) {
+          return;
+        }
+        if (
+          player.spec.role !== ROLES.HEALER &&
+          player.spec.role !== ROLES.TANK &&
+          player.spec !== SPECS.AUGMENTATION_EVOKER
+        ) {
+          const i18nClassName = player.spec.className ? i18n._(player.spec.className) : '';
+          const className = i18nClassName?.replace(/\s/g, '') ?? '';
+
+          playerWhitelist.set(player.name, className);
+        }
+      });
+    });
+  }
+
+  /** Generate filter based on black list and whitelist */
+  getFilter() {
+    const playerNames = Array.from(playerWhitelist.keys());
+    const nameFilter = playerNames.map((name) => `source.name="${name}"`).join(' OR ');
```

Comment:
Double check that this includes pets. May need to also check `source.owner.name`

Reply:
It does include pet damage, yea.

Reply:
(which we want)

Reply:
👍 haven't gotten to read through it yet but this looks really useful. i should get through PR reviews tomorrow

Reply:
Cool, just dont merge it until I’ve made one more change, thought of a bug that can occour in the way I manage the damage data that I need to fix first :x